siriuspy/siriuspy/clientarch/pvarch.py

```diff
@@ -52,22 +63,127 @@ def update(self):
             elif field == 'Host name':
                 self.host_name = value
             elif field == 'Is this PV currently connected?':
-                self.connected = (value.lower() == 'yes')
+                self.is_connected = (value.lower() == 'yes')
             elif field == 'Average bytes per event':
+                value = value.replace(',','.')
                 self.avg_bytes_per_event = float(value)
             elif field == 'Estimated storage rate (KB/hour)':
+                value = value.replace(',', '.')
                 self.estimated_storage_rate_kb_hour = float(value)
             elif field == 'Estimated storage rate (MB/day)':
+                value = value.replace(',', '.')
                 self.estimated_storage_rate_mb_day = float(value)
             elif field == 'Estimated storage rate (GB/year)':
+                value = value.replace(',', '.')
                 self.estimated_storage_rate_gb_year = float(value)
+        return True
+
+    def __str__(self):
+        """."""
+        rst = ''
+        rst += '{:<30s}: {:}\n'.format('pvname', self.pvname)
+        rst += '{:<30s}: {:}\n'.format('is_scalar', self.is_scalar)
+        rst += '{:<30s}: {:}\n'.format('is_paused', self.is_paused)
+        rst += '{:<30s}: {:}\n'.format('is_connected', self.is_connected)
+        rst += '{:<30s}: {:}\n'.format('nelms', self.nelms)
+        rst += '{:<30s}: {:}\n'.format('units', self.units)
+        rst += '{:<30s}: {:}\n'.format('host_name', self.host_name)
+        rst += '{:<30s}: {:}\n'.format(
+            'avg_bytes_per_event', self.avg_bytes_per_event)
+        rst += '{:<30s}: {:}\n'.format(
+            'estimated_storage_rate_kb_hour',
+            self.estimated_storage_rate_kb_hour)
+        rst += '{:<30s}: {:}\n'.format(
+            'estimated_storage_rate_mb_day',
+            self.estimated_storage_rate_mb_day)
+        rst += '{:<30s}: {:}\n'.format(
+            'estimated_storage_rate_gb_year',
+            self.estimated_storage_rate_gb_year)
+        return rst
+
+
+class PVData:
+    """PV Arch Details."""
+
+    def __init__(self, pvname, connector=None):
+        """."""
+        self.pvname = pvname
+        self.connector = connector
+        self.timestamp_start = None
+        self.timestamp_stop = None
+        self.timestamp = None
+        self.value = None
+        self.status = None
+        self.severity = None
+
+    @property
+    def request_url(self):
+        """."""
+        url = self.connector.getData(
```

Comment:
there is a `self.connect()` missing here.

Reply:
done